src/MongoDB.Analyzer/Core/Builders/BuildersResolveVariables.cs

```diff
@@ -0,0 +1,379 @@
+        var analysisContexts = buildersAnalysis.AnalysisContext;
+        var dirtyVariables = nodeContext.DirtyVariables;
+        var syntaxKind = syntaxNode.Kind();
+
+        if (RHS is VariableDeclaratorSyntax variableDeclaratorSyntax)
+        {
+            var identifier = variableDeclaratorSyntax.Identifier;
+            var variableName = identifier.ValueText;
+            variableNames.Add(variableName);
+            RHS = variableDeclaratorSyntax.Initializer.Value;
+        }
+        else if (RHS is AssignmentExpressionSyntax)
+        {
+            syntaxKind = RHS.Kind();
+            while (RHS is AssignmentExpressionSyntax assignmentExpression)
+            {
+                var left = assignmentExpression.Left;
+                var identifierNodes = left.DescendantNodesAndSelf().OfType<IdentifierNameSyntax>();
+                foreach (var identifier in identifierNodes)
+                {
+                    var variableName = identifier.Identifier.ValueText;
+                    variableNames.Add(variableName);
+                }
+                RHS = assignmentExpression.Right;
+            }
+        }
+        
+        var argumentTypeName = "";
+        var nodesRemapping = new Dictionary<SyntaxNode, SyntaxNode>();
+        var constantMappers = new List<ConstantsMapper>();
+
+        foreach (var builderNode in buildersNodes)
+        {
+            SyntaxNode rewrittenBuildersNode = null;
+            ConstantsMapper constantsMapper = new ConstantsMapper();
+            if (builderNode is IdentifierNameSyntax identifierName)
+            {
+                var analysisContext = variableValues[identifierName.Identifier.ValueText];
+                rewrittenBuildersNode = analysisContext.Node.RewrittenExpression;
+                constantsMapper = analysisContext.Node.ConstantsRemapper;
+                argumentTypeName = analysisContext.Node.ArgumentTypeName;
+                ExpressionAnalysisContext diagnostic = new ExpressionAnalysisContext(new ExpressionAnalysisNode(builderNode, argumentTypeName, rewrittenBuildersNode, constantsMapper));
+                analysisContexts.Add(diagnostic);
+            }
+            else
+            {
+                var analysisContext = buildersToExpressionContext[builderNode];
+                rewrittenBuildersNode = analysisContext.Node.RewrittenExpression;
+                constantsMapper = analysisContext.Node.ConstantsRemapper;
+                argumentTypeName = analysisContext.Node.ArgumentTypeName;
+            }
+            nodesRemapping.Add(builderNode, rewrittenBuildersNode);
+            constantMappers.Add(constantsMapper);
+        }
+
+        SyntaxNode buildersExpression = null;
+        if(RHS is IdentifierNameSyntax identifierNameSyntax)
+        {
+            buildersExpression = variableValues[identifierNameSyntax.Identifier.ValueText].Node.RewrittenExpression;
+        }
+        else if(RHS is InvocationExpressionSyntax invocationExpression)
```

Comment:
pay attention to all spaces if_space_(...

Reply:
Fixed

Reply:
Not everywhere